### PR TITLE
Check the DB list of validations to determine validation set success status

### PR DESCRIPTION
**Pull request by agr**
Addresses https://github.com/NuGet/Engineering/issues/1022

Now, only the validations from DB are considered for determining validation set success.

### CODE REVIEW

File: src/NuGet.Services.Validation.Orchestrator/ValidationOutcomeProcessor.cs [hunk 1]
```diff
@@ -138,15 +138,8 @@ public class ValidationOutcomeProcessor : IValidationOutcomeProcessor
...
-                .PackageValidations
-                .Where(v => v.ValidationStatus == ValidationStatus.Succeeded)
-                .Select(v => v.Type));
-            var requiredValidations = _validationConfiguration
-                .Validations
-                .Select(v => v.Name);
-
-            return completeValidations.IsSupersetOf(requiredValidations);
+            var allSucceeded = packageValidationSet.PackageValidations.All(pv => pv.ValidationStatus == ValidationStatus.Succeeded);
+            return allSucceeded;
```
Comment (scottbommarito) [inline src/NuGet.Services.Validation.Orchestrator/ValidationOutcomeProcessor.cs]: nit: no need to save in a variable if you just return it immediately afterwards

Comment (loic-sharma) [inline src/NuGet.Services.Validation.Orchestrator/ValidationOutcomeProcessor.cs]: >allSucceeded [](start = 19, length = 12)

Nit: return directly? :)

File: tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationOutcomeProcessorFacts.cs [hunk 2]
```diff
@@ -227,6 +227,42 @@ public class ValidationOutcomeProcessorFacts
                 .Verify(ps => ps.UpdatePackageStatusAsync(Package, PackageStatus.FailedValidation, It.IsAny<bool>()), Times.Never());
         }
 
+        [Fact]
+        public async Task PrefersDbOverConfigurationForDeterminingSuccess()
```
Comment (scottbommarito) [inline tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationOutcomeProcessorFacts.cs]: So it appears that here we have a test for:
If there is less validations in DB than in configuration, and all of the DB validations have succeeded, we return true.

I think we should also consider
1 - MORE DB validations than configuration and all succeeded
2 - MORE DB validations than configuration and some/all have failed
3 - LESS DB validations than configuration and some/all have failed

Comment (agr) [reply tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationOutcomeProcessorFacts.cs]: First two cases have their own issue and are out of scope.

Comment (scottbommarito) [reply tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationOutcomeProcessorFacts.cs]: Ok--should still add a test for 3.

Comment (agr) [reply tests/NuGet.Services.Validation.Orchestrator.Tests/ValidationOutcomeProcessorFacts.cs]: Issue with 3 is that with any failed validation the `AllValidationsSucceeded` would not even be called... 
Added anyway.

**Pull request merged**